### PR TITLE
Draggable feedback positioning

### LINKED ISSUE #145639: Draggable Feedback is positioned incorrectly when the Overlay is scaled
### Steps to reproduce

1. Wrap your entire `MaterialApp` in a transform (`Transform.scale(scale: 0.5, child: ...)` works well) 
2. Try out any `Draggable`

### Expected results

The `Draggable` feedback should still be attached to the pointer, as if the App wasn't transformed at all.

### Actual results

The feedback is offset, because it is positioned in the respective overlay using global coordinates.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'dart:math';

import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return ColoredBox(
      color: Colors.grey[100]!,
      child: Transform.rotate(
        angle: 0.1 * pi,
        child: Transform.scale(
          scale: 0.5,
          child: Material(
            color: Colors.grey[900],
            elevation: 8,
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(
                25,
              ),
            ),
            child: Padding(
              padding: const EdgeInsets.all(50.0),
              child: MaterialApp(
                title: 'Flutter Demo',
                debugShowCheckedModeBanner: false,
                theme: ThemeData(
                  primarySwatch: Colors.blue,
                ),
                home: const MyHomePage(title: 'Flutter Demo Home Page'),
              ),
            ),
          ),
        ),
      ),
    );
  }
}

class MyHomePage extends StatefulWidget {
  final String title;

  const MyHomePage({
    super.key,
    required this.title,
  });

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Draggable(
              child: FlutterLogo(),
              feedback: FlutterLogo(),
            ),
            Text("Drag this Logo"),
          ],
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/flutter/flutter/assets/42270125/0320b6a7-8a30-4d84-acd1-68ceae43ef78

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[!] Flutter (Channel master, 3.21.0-13.0.pre.24, on macOS 14.3.1 23D60 darwin-arm64, locale en-DE)
    • Flutter version 3.21.0-13.0.pre.24 on channel master at /Users/tim/Developer/whynotmake.it/flutter
    ! Upstream repository https://github.com/whynotmake-it/flutter.git is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to https://github.com/whynotmake-it/flutter.git to dismiss this error.
    • Framework revision 73e78fd97c (4 hours ago), 2024-03-23 10:18:25 -0400
    • Engine revision 689ea4b10b
    • Dart version 3.4.0 (build 3.4.0-265.0.dev)
    • DevTools version 2.34.0-dev.12
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc2)
    • Android SDK at /Users/tim/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0-rc2
    • Java binary at: /Library/Java/JavaVirtualMachines/zulu-11.jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment Zulu11.54+25-CA (build 11.0.14.1+1-LTS)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    ✗ Unable to get list of installed Simulator runtimes.
    • CocoaPods version 1.13.0

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.87.0-insider)
    • VS Code at /Applications/Visual Studio Code - Insiders.app/Contents
    • Flutter extension version 3.84.0

[✓] Connected device (2 available)
    • macOS (desktop)                 • macos                 • darwin-arm64 • macOS 14.3.1 23D60 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin       • macOS 14.3.1 23D60 darwin-arm64
    ! Error: Browsing on the local area network for iPhone von mir. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 4 categories.
```

</details>

### PR DESCRIPTION
**Pull request by timcreatedit**
Reopened after revert in #147658

Another test was added in the meantime to `draggable_test.dart` that didn't call `gesture.up()`. I added this call to the test and now all tests pass.

---
# Original Description (#145647):
We changed the coordinates used to position the `Draggable` feedback by transforming them into the `Overlay`s coordinate space. This has no influence on any untransformed `Overlay`, most Flutter apps should be not affected. 
This PR fixes the positioning of the feedback in transformed context (see #145639 for before video):

https://github.com/flutter/flutter/assets/42270125/df34e198-0667-453d-a27a-a79b2e2825a1

- fixes #145639 

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Review by dkwingsmt — APPROVED** [review]
LGTM!

File: packages/flutter/test/widgets/draggable_test.dart [hunk 1]
```diff
@@ -3497,6 +3529,93 @@ void main() {
...
+    final Offset secondLocation = location + const Offset(100, 100);
+    await gesture.moveTo(secondLocation);
+    await tester.pump();
+    final Offset appTopLeft = tester.getTopLeft(find.byType(MaterialApp));
+    expect(tester.getTopLeft(find.text('Source')), appTopLeft);
+    expect(tester.getTopLeft(find.text('Feedback')), secondLocation);
+
+    // Finish gesture to release resources.
+    await gesture.up();
+    await tester.pumpAndSettle();
```
Comment (justinmc) [inline packages/flutter/test/widgets/draggable_test.dart:3557]: Nit: Should this be `pump`? Just noticing that it's usually  `pump` elsewhere. Same for the two new tests below.

**Review by justinmc — APPROVED** [review]
LGTM :+1:. Thanks for following up after the revert.

I wish we had an automated way to catch this lack of `.up` calls. I was just asking @polina-c about this.

File: packages/flutter/test/widgets/draggable_test.dart (hunk 1 above)
Comment (timcreatedit) [reply packages/flutter/test/widgets/draggable_test.dart]: Good point, works just as well with `pump`! I think I just copied the three lines from the test right above the ones I added without thinking too much about it. Fixed!

**Comment by dkwingsmt** [comment]
Ok, I've tested that this will trigger error if the root widget (which supposedly provides the overlay) is take away during the drag.

Reproduction:
1. Apply this PR (or simply access the context during `_DragAvatar.updateDrag`
```dart
  void updateDrag(Offset globalPosition) {
    final RenderBox box = overlayState.context.findRenderObject()! as RenderBox;
    final Offset overlaySpaceOffset = box.globalToLocal(globalPosition);
    print(overlaySpaceOffset - dragStartPoint);
```
2. Run the following app:
```dart
import 'package:flutter/material.dart';

class Main extends StatefulWidget {

  @override
  _MainState createState() => _MainState();
}

class _MainState extends State<Main> {
  bool _show = true;

  @override
  Widget build(BuildContext context) {
    return !_show ? Container() : MaterialApp(
      home: ListView(
        scrollDirection: Axis.horizontal,
        children: <Widget>[
          DragTarget<int>(
            builder: (BuildContext context, List<int?> data, List<dynamic> rejects) {
              return const Text('Target');
            },
            onAcceptWithDetails: (DragTargetDetails<int> _) {
            },
          ),
          Container(width: 400.0),
          Draggable<int>(
            data: 1,
            feedback: Text('H'),
            onDragStarted: () {
              Future<void>.delayed(Duration(seconds: 2)).then((_) {
                setState(() {
                  _show = false;
                });
              });
            },
            childWhenDragging: SizedBox(),
            axis: Axis.horizontal,
            child: Text('H'),
          ),
        ],
      ),
    );
  }
}

void main() {
  runApp(Main());
}
```
3. Drag the `H` and hold for 2 seconds before moving

Throws error:
```

══╡ EXCEPTION CAUGHT BY GESTURE LIBRARY ╞═══════════════════════════════════════════════════════════
The following assertion was thrown while routing a pointer event:
This widget has been unmounted, so the State no longer has a context (and should be considered
defunct).
Consider canceling any active work during "dispose" or using the "mounted" getter to determine if
the State is still active.

When the exception was thrown, this was the stack:
#0      State.context.<anonymous closure> (package:flutter/src/widgets/framework.dart:951:9)
#1      State.context (package:flutter/src/widgets/framework.dart:957:6)
#2      _DragAvatar.updateDrag (package:flutter/src/widgets/drag_target.dart:857:40)
#3      _DragAvatar.update (package:flutter/src/widgets/drag_target.dart:839:5)
#4      MultiDragPointerState._move (package:flutter/src/gestures/multidrag.dart:100:16)
#5      MultiDragGestureRecognizer._handleEvent (package:flutter/src/gestures/multidrag.dart:267:13)
#6      PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:98:12)
#7      PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:143:9)
#8      _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:633:13)
#9      PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:141:18)
#10     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:127:7)
#11     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:495:19)
#12     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:475:22)
#13     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:450:11)
#14     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:420:7)
#15     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:383:5)
#16     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:330:7)
#17     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:299:9)
#18     _invoke1 (dart:ui/hooks.dart:328:13)
#19     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:442:7)
#20     _dispatchPointerDataPacket (dart:ui/hooks.dart:262:31)

router: Instance of 'PointerRouter'
route: Closure: (PointerEvent) => void from Function '_handleEvent@171470698':.
event: PointerMoveEvent#dc09b(position: Offset(379.5, 32.1))
════════════════════════════════════════════════════════════════════════════════════════════════════

Another exception was thrown: This widget has been unmounted, so the State no longer has a context (and should be considered defunct). 
```

We should probably fix this error.

**Review by dkwingsmt — CHANGES_REQUESTED** [review]
See comment

**Comment by timcreatedit** [comment]
Thanks for finding this, that's very insightful. As I mentioned on the other PR it didn't quite sit right with me that the test failures were triggered by the changes. I will take a look at this again!

**Comment by timcreatedit** [comment]
@dkwingsmt I added a new test that verifies the behaviour you found when unmounting, and added a mounted check to the context access in `_DragAvatar.updateDrag`. Now the test passes. Do you want me to revert the changes I made to other tests, or do we still want to finish the gestures there? I reckon since unmounting is now covered by a test, we might as well leave it in?

Thanks again for your help!

**Review by dkwingsmt — APPROVED** [review]
LGTM, thanks for working on this!

**Comment by dkwingsmt** [comment]
I would prefer removing the `gesture.up`s in existing tests since they are unneeded changes (for the PR) and unneeded lines (for the tests), but I'm not strong on this.

Awesome test for the graceful unmounting, BTW!

**Review by justinmc — APPROVED** [review]
LGTM, but see the analyzer failure.

**Comment by timcreatedit** [comment]
> I would prefer removing the `gesture.up`s in existing tests since they are unneeded changes (for the PR) and unneeded lines (for the tests), but I'm not strong on this.
> 
> 
> 
> Awesome test for the graceful unmounting, BTW!

Removed them all! Thanks :)

**Comment by timcreatedit** [comment]
> LGTM, but see the analyzer failure.

@justinmc Ready, it's all green 🌿

**Comment by timcreatedit** [comment]
@dkwingsmt @justinmc Just wanted to touch base again, is there anything you still need from me?

**Comment by dkwingsmt** [comment]
Nope. Here we go! Thanks for the work!

**Pull request merged**